editor/scene_tree_dock.cpp

```diff
@@ -2854,11 +2878,6 @@ void SceneTreeDock::_tree_rmb(const Vector2 &p_menu_pos) {
 	menu->popup();
 }
 
-void SceneTreeDock::_update_tree_menu() {
-	PopupMenu *tree_menu = button_tree_menu->get_popup();
-	tree_menu->set_item_checked(tree_menu->get_item_idx_from_text(TTR("Auto Expand to Selected")), EditorSettings::get_singleton()->get("docks/scene_tree/auto_expand_to_selected"));
```

Comment:
This removal makes the options no longer synced with editor settings. You should ensure that they are updated on startup and when changing the setting from Editor Settings dialog.